web/react/components/mention_list.jsx

```diff
@@ -118,17 +213,21 @@ module.exports = React.createClass({
 
             if (firstName.lastIndexOf(mentionText,0) === 0
                     || lastName.lastIndexOf(mentionText,0) === 0 || users[i].username.lastIndexOf(mentionText,0) === 0) {
-                mentions[i+1] = (
+                mentions[index] = (
                     <Mention
                         ref={'mention' + index}
                         username={users[i].username}
                         secondary_text={users[i].secondary_text}
                         id={users[i].id}
+                        listId={index}
+                        isFocus={this.state.selectedMention === index ? true : false}
+                        handleMouseEnter={this.handleMouseEnter}
```

Comment:
If you do this instead:

```handleMouseEnter={function(value) { return function() { self.handleMouseEnter(value); } }(index)}```

Then you shouldn't need to pass in listId to the mention control